src/common/gui/SurgeGUIEditor.cpp

```diff
@@ -3254,31 +3254,68 @@ int32_t SurgeGUIEditor::controlModifierClicked(CControl* control, CButtonState b
                   r->setChecked( true );
                eid++;
 
-               if( p->ctrlgroup == cg_LFO )
+               if (p->ctrlgroup == cg_ENV || p->ctrlgroup == cg_LFO)
                {
-                  char lab[256];
-                  char prefix[32];
+                  char label[256];
+                  char prefix[128];
+                  char pars[32];
 
-                  // WARNING - this won't work with Surge++
-                  int a = p->ctrlgroup_entry + 1 - ms_lfo1;
-                  if (a > 6)
-                     sprintf(prefix, "SLFO %i", a - 6);
-                  else
-                     sprintf(prefix, "LFO %i", a);
+                  int a = p->ctrlgroup_entry;
+
+                  switch (a)
+                  {
+                  case 0:
+                  {
```

Comment:
Thisn0 and 1 should be the modulator name not the content 0:or 1, no?

Reply:
Oh no it just needs a comment. Got it.

Reply:
OK so stylistically doing this as a switch where we mix the different entries of different control groups is what got me.

I think this code should look like

```
if( ctrglr == LFO )
   if( entry > 6 )
      SLFO
   else
     VLFO
elseif( ctrlgrp == ENV )
   if( entry == 0 )
     AEG
   else
     FEG
```

in pseudocode. Switching over those values is dangerous (what if we re-define ms_lfo1 to 0 - this would break) since they have different meanings in different context.